### ISSUE TITLE
Platform Compat Analyzer should recognize supported platforms in the inconsistent list scenario

### ISSUE THREAD

**Issue by jeffhandley** [dotnet/roslyn-analyzers#4168]
### Analyzer

**Diagnostic ID**: [CA1416](https://docs.microsoft.com/visualstudio/code-quality/ca1416): `Validate platform compatibility`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: 5.0.100-rc.2.20467.7 ([daily build](https://github.com/dotnet/installer))

### Describe the bug

When the platform compat analyzer encounters an "inconsistent list" scenario, it favors the "deny list" (unsupported platforms).  This is yielding false negatives and it should be inverted.

### Steps To Reproduce

Many annotations have been applied in the .NET Libraries for RC2 that mark APIs as unsupported on "browser".  Some of these annotations have been applied at the assembly-level, such as in the `System.Diagnostics.Process` assembly.  There were previously APIs marked as supported only on "windows", including some APIs within that same assembly.

In a `net5.0` project using the RC2 SDK, include the following line of code:

```csharp
System.Diagnostics.Process.GetCurrentProcess().MaxWorkingSet = new IntPtr(5);
```

### Expected behavior

There should be a diagnostic warning produced that the `MaxWorkingSet` property setter is supported on windows.

### Actual behavior

No diagnostic is produced.

## Additional context

If you update the project file to indicate that the project supports "browser" through `<SupportedPlatform Include="browser" />`, then a diagnostic warning is produced indicating that `MaxWorkingSet` is unsupported on browser.  This confirms that the analyzer is currently favoring the "deny list" (unsupported platforms).  We were previously stating that this inconsistent list would have the deny list behavior but that the scenario is unsupported.  Since we have produced the scenario ourselves, we should treat this scenario as supported and favor the allow list (supported platforms).

**Comment by buyaa-n** [comment]
Fixed by https://github.com/dotnet/roslyn-analyzers/pull/4208
